pika/channel.py

```diff
@@ -417,20 +489,46 @@ def basic_recover(self, callback=None, requeue=False):
                          [spec.Basic.RecoverOk])
 
     def close(self, reply_code=0, reply_text="Normal Shutdown"):
-        """Will invoke a clean shutdown of the channel with the AMQP Broker.
+        """Invoke a graceful shutdown of the channel with the AMQP Broker.
+
+        If channel is OPENING, transition to CLOSING and suppress the incoming
+        Channel.OpenOk, if any.
 
-        :param int reply_code: The reply code to close the channel with
-        :param str reply_text: The reply text to close the channel with
+        :param int reply_code: The reason code to send to broker
+        :param str reply_text: The reason text to send to broker
 
+        :raises ChannelClosed: if channel is already closed
+        :raises ChannelAlreadyClosing: if channel is already closing
         """
-        if not self.is_open:
-            raise exceptions.ChannelClosed()
-        LOGGER.info('Channel.close(%s, %s)', reply_code, reply_text)
-        if self._consumers:
-            LOGGER.debug('Cancelling %i consumers', len(self._consumers))
-            for consumer_tag in dictkeys(self._consumers):
```

Comment:
Why the tuple and trailing comma? Creating an extra data structure for the formatting seems unnecessary.